hygen/_templates/source/new/prompt.js

```diff
@@ -0,0 +1,71 @@
+// see types of prompts:
+// https://github.com/enquirer/enquirer/tree/master/examples
+//
+const { promptArgs, doesFileExists } = require('../../promptUtils');
+
+const SOURCE_TYPES = ['sql', 'bq'];
+
+const LAYER_TYPES = {
+  [SOURCE_TYPES[0]]: {
+    title: 'SQL dataset',
+    msg: 'Type a query',
+  },
+  [SOURCE_TYPES[1]]: {
+    title: 'BigQuery Tileset',
+    msg: 'Type the name of your tileset',
+  },
+};
+
+const prompt = async ({ prompter, args }) => {
+  let questions = [];
+
+  if (!args.name) {
+    questions.push({
+      type: 'input',
+      name: 'name',
+      message: 'Name:',
+    });
+  }
+
+  // Check name to remove source word if the user added it by (his/her)self
+  let answers = await promptArgs({ prompter, args, questions });
+  answers.name = answers.name.replace('Source', '').replace('source', '');
+
+  // Check if source already exists
+  const fileName = [answers.name[0].toUpperCase(), ...answers.name.slice(1, answers.name.length)].join('')
+  const doesSourceAlreadyExists = await doesFileExists(['src', 'data', 'sources', `${fileName}Source.js`]);
+  if (doesSourceAlreadyExists) throw new Error(`The source ${answers.name} already exists.`);
```

Comment:
If source already exists, the code is executed but nothing happens because hygen has the `skip` flag if that source exists.

Reply:
Fixed.